source

```
@@ -90444,8 +90453,11 @@ interface <dfn>ImageBitmapFactories</dfn> {
      official type.</p></li>
 
      <li><p>If the image data is not in a supported bitmap image file format (e.g. it's a vector
-     graphic or it's not an image at all), or if the image data is corrupted in some fatal way such
-     that the image dimensions cannot be obtained, then reject the promise with null, and abort
+     graphic or it's not an image at all), then reject the promise with an
+     <code>InvalidStateError</code> exception, and abort these steps.</p></li>
+
+     <li><p>If the image data is corrupted in some fatal way such that the image dimensions cannot
```

In the section "If <var>image</var> is an <code>img</code> element", should we have something equivalent to this? Example:  If the img is in the broken state (data-x="img-error"), the promise should be rejected with an <code>EncodingError</code> exception. 

Right now, the the broken state case gets handled by a catch-all clause for handling imgs that are not in the completely available state. It rejects with an InvalidStateError.